### ISSUE TITLE
Inconsistent treatment of reference to another module

### ISSUE THREAD

**Issue by paulch42** [overturetool/overture#654]
### Description

If a reference is made to a function or value from a different module using the fully qualified name but the module is not imported, an error is raised.
If a reference is made to a type from a different module using the fully qualified name but the module is not imported, no error is raised.

### Steps to Reproduce
Module `B`.
```
module B
exports all
definitions 

types
  B = nat;
values
  bval:B = 1;
functions
  f:B->B
  f(b) == b+1;

end B
```
Module `A` that uses type `B`, function `f` and value `bval` from module `B`.
```
module A
--imports from B all
exports all
definitions

values
  aval:B`B = B`bval;
functions
  g:B`B->B`B
  g(b) == B`f(b);

end A
```
Overture reports the references to function `f` and value `bval` as errors, but allows the reference to type `B`. Note import of `B` is commented out. Behaviour is consistent when the import is included.

**Expected behaviour:**
Either references to both the type and the function/value are accepted, or all are flagged as an error.
Which of the two should it be?

**Actual behavior:**
Type reference is accepted, function/value reference is flagged as an error.

**Reproduces how often:**
Always.

### Versions
Overture 2.5.4, MacOS 10.13.1.

### Additional Information
None.

**Comment by nickbattle** [comment]
Yes, I think this may have been noticed before. I'll check, buy I can guess the underlying cause... When  type checking is performed, one of the first actions is to resolve the types across the whole specification, so that every type can be chased down to its primitives. To do that, VDMJ ignores the module scope initially because it is trying to get to the "full picture". I suspect the results of this early pass are leaking through to the subsequent scoped checks of the other definitions.

It should be possible to limit the type definitions to just those imported before the scoped checks are made.
